core/src/main/scala/cats/FlatMap.scala

```diff
@@ -90,4 +91,17 @@ import simulacrum.typeclass
    */
   def ifM[B](fa: F[Boolean])(ifTrue: => F[B], ifFalse: => F[B]): F[B] =
     flatMap(fa)(if (_) ifTrue else ifFalse)
+
+  /**
+   * Keeps calling `f` until a `[[cats.data.Xor.Right Right]][B]` is returned.
+   *
+   * Based on Phil Freeman's
+   * [[http://functorial.com/stack-safety-for-free/index.pdf Stack Safety for Free]].
+   *
+   * Implementations of this method must use constant stack space.
```

Comment:
We should remove this comment -- implementations are no longer required to do this.